src/Ogooreck.Sample.BusinessLogic.FSharp.Tests/Deciders/BankAccountTests.fs

```diff
@@ -0,0 +1,115 @@
+module BankAccountTests
+
+open System
+open Deciders.BankAccount
+open Deciders.BankAccountPrimitives
+open FsCheck
+open Ogooreck.BusinessLogic
+open Deciders.BankAccountDecider
+open FsCheck.Xunit
+
+let random = Random()
+let now = DateTimeOffset.UtcNow
+
+let spec =
+    Specification.For(decide, evolve, (fun () -> NotInitialised))
+
+let BankAccountOpenedWith =
+    fun bankAccountId now version ->
+        let accountNumber =
+            AccountNumber.newNumber (Guid.NewGuid().ToString())
+
+        let clientId = ClientId.newId ()
+
+        let currencyISOCode =
+            CurrencyCode.newCode "USD"
+
+        BankAccountOpened
+            { BankAccountId = bankAccountId
+              AccountNumber = accountNumber
+              ClientId = clientId
+              CurrencyISOCode = currencyISOCode
+              CreatedAt = now
+              Version = version }
+
+let BankAccountClosedWith =
+    fun bankAccountId now version ->
+        BankAccountClosed
+            { BankAccountId = bankAccountId
+              Reason = Guid.NewGuid().ToString()
+              ClosedAt = now
+              Version = version }
+
+[<Property>]
+let ``GIVEN non existing bank account WHEN open with valid params THEN bank account is opened``
+    bankAccountId
+    accountNumber
+    clientId
+    currencyISOCode
+    =
+    let notExistingAccount = Array.empty<_>
```

Comment:
I believe that the `<_>` can be omitted.

Reply:
That's correct 👍 Updated in https://github.com/oskardudycz/Ogooreck/pull/13/commits/9c3bd4fa7efd66b75bf111513ba31eea4892e0cb.